### DISCUSSION TITLE
1.0.99997 | Alpha 4.4 for 1.1 | Current: Version 1.1-A4.4

### DISCUSSION THREAD

**Discussion by cyberofficial** [cyberofficial/Synthalingua#122] [Announcements]
Portable + GUI is at itch.io! Pick it up there.
[<img src="https://i.imgur.com/dyZz6u5.png" width=50%>](https://cyberofficial.itch.io/synthalingua)

-- Please allow some time for itch.io to process the update --

---------------------------
Version 1.1-A 4.4

This update focuses on enhancing the user experience for set up and portable issue, fixing bugs, improving compatibility, and simplifying the setup process.

**Added:**

* **New application icon** Synthalingua GUI now has a distinctive icon.
* **Automatic dependency setup:** A new script, `set_up_env.py`, automates the download and extraction of FFmpeg and yt-dlp, simplifying the initial setup process. This also has a windows exe file in the portable version.

**Changed:**

* **Minimum Supported Windows Version:** The portable version of the GUI now requires Windows 10 (build 10.0.17763 or later) due to a change in .NET framework requirements. Windows 7 is no longer supported for the portable GUI. 
* **Code of Conduct:** The project's Code of Conduct has been updated to be more comprehensive and inclusive, promoting a welcoming and respectful community. 
* **Updated `setup.bat` Script:** Improved the `setup.bat` script to handle Python installations not found in the system's PATH and to provide clearer instructions to users.

**Fixed:**
* **Batch Script for FFmpeg Path:** The batch script responsible for setting the FFmpeg path has been rewritten for clarity and efficiency.
* Fixed #119 
* Fixed #121 

**Removed:**
* **Manual FFmpeg/yt-dlp Setup:**  Users no longer need to manually download and extract FFmpeg, 7zr, or yt-dlp. The new `set_up_env.py` script handles this automatically.
* **Outdated PyTorch CUDA Packages:** The setup script no longer installs CUDA-specific PyTorch packages, preventing potential compatibility issues.

We are committed to improving Synthalingua and welcome feedback and contributions from the community.

<hr /><em>This discussion was created from the release <a href='https://github.com/cyberofficial/Synthalingua/releases/tag/1.0.99995a44'>1.0.99997 | Alpha 4.4 for 1.1 | Current: Version 1.1-A4.4</a>.</em>

**Comment by SteveMurgaski** [comment]
I downloaded the full Windows version (not the GUI one) from itch.io, but the instructions in the Github ReadMe and on the Synthalingua Wiki don't seem to make sense. There's no setup.bat in the zip from twitch.io. Does running set_up_env.exe replace running setup.bat? The command examples on the Wiki don't work because there are no .py files in the zip.
Sorry but I'm really lost. What instructions should I follow to install and use this .zip file?

**Reply by cyberofficial** [reply]
Hello! The portable version is portable, means no need to install stuff manually.

When you download from itch, make sure to grab the one that says
```
[ALPHA] Synthalingua Windows x64 Portable
3.4 GB
Version 1.1-A4.6-GUI4.45
```

the `setup.bat` is for when you download the repo as is, and already have python, ffmpeg, and other dependencies installed. To run the program from source. The `setup.bat` is more for the advanced users who download the repo and mess with it that way.

When you download the portable and unzip it, run the `SynthalinguaGUI.exe` program. This will make using it as easy as possible. The `set_up_env.exe` program will automatically run with the GUI program, it will run only once.

**Reply by SteveMurgaski** [reply]
Thank you, that helped. I don't have it working yet though. (Should I start an issue? I'm new to github.)
I can't use the GUI because I use screen reader software, JAWS for Windows, which doesn't work well with most GUIs. So I downloaded the repo. But I keep getting errors.
When trying to generate translated subtitles for a .flac audio file, I get:

Loading Primary Imports
Error Loading Primary Imports
Check to make sure you have all the required modules installed.
Error: No module named ’speech_recognition’
Error Loading Primary Imports
Check the Modules folder for the imports.py file and make sure it is not missing or corrupted.
name ’sys’ is not defined
Traceback (most recent call last):
File "C:\Users\Steve\Utils\Synthalingua\modules\imports.py", line 6, in <module>
import speech_recognition as sr
ModuleNotFoundError: No module named ’speech_recognition’
[...]

The imports.py file is in the modules folder.
I've followed the installation instructions in the Github readme. I tried running setup.bat a few times to see if it could fix any missing modules, but continue getting this error.
I also tried running the transcribe_audio.exe file from the itch.io .zip file, using command line arguments copied from the readme examples (replacing "python transcribe_audio" with "transcribe_audio.exe" while keeping other arguments the same.) That gave me slightly different errors, but still didn't work.
I hope you can help. I'm excited about this program.

**Reply by cyberofficial** [reply]
The error you see is because you are not running in a python environment.

What happens when you run `setup.bat` since you downloaded the repo?

**Comment by SteveMurgaski** [comment]
Setup.bat says that "Python environment ’data_whisper’ already exists.
Reinstall it? [Y/N]:". Do I start this virtual environment and then run the
programs?

**Reply by cyberofficial** [reply]
was there any error during the setup process? You should try to reinstall and see if there is any error.

**Comment by SteveMurgaski** [comment]
I reinstalled using setup. The only thing that looks like an error is:
"WARNING: Skipping torchvision as it is not installed.
WARNING: Skipping torchaudio as it is not installed."
Those show up every time I run setup.bat.
I also tried running activate.bat in the data_whisper\scripts dir, and then
running the sample commands from the Github readme in the virtual
environment. That got much further than before, but after the message like
"Get a coffee or something, this could take awhile," it started giving
errors and stopped.

**Reply by cyberofficial** [reply]
If you still have the itch.io version

Could you try doing "set_up_env.exe" by it self like so, allow it to install the required files, then try doing these commands in order:

```
$> set_up_env.exe
<follow the onscreen prompts>

call ffmpeg_path.bat


transcribe_audio.exe --ram 12gb-v3 --makecaptions --file_input "replaceme.mp4" --file_output "%cd%" --file_output_name "replaceme" --language ja --device cuda --condition_on_previous_text 

cuda --condition_on_previous_text
```

Replace `replaceme.mp4` with your input file `--file_output "%cd%"` can be left as `"%cd%"` which will keep it in the same location as the exe. `--file_output_name` is usually the file input name minus the ext. so like `"some video about apples.mp4"` will be `"some video about apples"`. `--condition_on_previous_text` helps prevents errors which increases time to make captions, but improves accuracy. Language can be what ever the main file language is. and Ram flag what ever you gpu can handle.

video:

https://github.com/user-attachments/assets/32307a46-30cf-4741-b903-a11ce51c3ea9

**Comment by SteveMurgaski** [comment]
I followed those instructions, but still got:
Error Loading Primary Imports
Check to make sure you have all the required modules installed.
Error: [WinError 126] The specified module could not be found. Error
loading "C:\Users\Steve\AppData\Local\Temp\_MEI9304
2\torch\lib\fbgemm.dll" or one of its dependencies.
Error Loading Primary Imports
Check the Modules folder for the imports.py file and make sure it is not
missing or corrupted.

I was running it on drive d: so I was a bit surprised it was looking for a
file in a temp folder on c. There is a c: drive on the system though.
My graphics card is only RTX 1050 so I don't think the --ram 12gb-v3 switch
would work, but I entered it as you wrote.
The language of my file is Spanish so I switched the --language switch to
es, instead of ja.

**Reply by cyberofficial** [reply]
Can you try this command in a admin command prompt or admin powershell?

```
winget install --id Microsoft.VisualStudio.2022.BuildTools --override "--passive --add Microsoft.VisualStudio.Component.VC.Tools.x86.x64"
```

Then trying again?

**Reply by cyberofficial** [reply]
I also released a new patch that changes somethings in the backend of the portalable, https://cyberofficial.itch.io/synthalingua please download "Version 1.1-A4.62-GUI4.45" and see if the issue is still present.

**Comment by SteveMurgaski** [comment]
I’ve started the download but it’s only going about 20 KB/sec. Last time the speed was more like 140 KB/sec, still bad but could work overnight. My connection is good but I’m in Philippines. I’ll keep trying but this could take a few days.On Oct 24, 2024, at 10:21 PM, Cyber Official ***@***.***> wrote:﻿
I also released a new patch that changes somethings in the backend of the portalable, https://cyberofficial.itch.io/synthalingua please download "Version 1.1-A4.62-GUI4.45" and see if the issue is still present.

**Comment by SteveMurgaski** [comment]
I keep getting the error "Failed - Forbidden" after downloading about 10%
of Synthalingua-alpha from itch.io. The download doesn't resume, it can
only be started over.

**Comment by SteveMurgaski** [comment]
Hi, just wanted to let you know that I tried to donate to Synthalingua, but got this error page:

We had an issue processing your payment

We're sorry about that. An error was encountered when processing your payment through our payment processor.

*	If your account has been charged please  <https://itch.io/support> contact support with the payment code P-26360120 and we can resolve the issue.
*	If you have not been charged then you can try again.
*	For any other questions or issues please  <https://itch.io/support> contact support.

Error Details

*	There was a problem executing your payment (201)

 <https://cyberofficial.itch.io/synthalingua> Return to Synthalingua

 <https://twitter.com/itchio> itch.io on Twitter <https://www.facebook.com/itchiogames> itch.io on Facebook

 <https://itch.io/docs/general/about> About <https://itch.io/docs/general/faq> FAQ <https://itch.io/blog> Blog <https://itch.io/support> Contact us

Copyright © 2025 itch corp ·  <https://itch.io/directory> Directory ·  <https://itch.io/docs/legal/terms> Terms ·  <https://itch.io/docs/legal/privacy-policy> Privacy ·  <https://itch.io/docs/legal/cookie-policy> Cookies

**Reply by cyberofficial** [reply]
Thanks for the consideration!

201 Means paypal placed a hold on your payment, Holds from paypal usually last 1 week. So either the money will go through on Oct 2, 2025 assuming you did it today, or you'll be refunded before 7 days.

I do other ways you can show support, such as ko-fi https://ko-fi.com/cyberofficial or through github directly https://github.com/sponsors/cyberofficial

**Reply by cyberofficial** [reply]
Hello, just incase you missed it, I did send you an email, please be sure to check it out.
